akka-typed-tests/src/test/scala/akka/typed/TimerSpec.scala

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.typed
+
+import scala.concurrent.duration._
+import scala.util.control.NoStackTrace
+
+import akka.typed.scaladsl.Actor._
+import akka.typed.scaladsl.AskPattern._
+import akka.typed.scaladsl.TimerScheduler
+import akka.typed.testkit.TestKitSettings
+import akka.typed.testkit.scaladsl._
+
+@org.junit.runner.RunWith(classOf[org.scalatest.junit.JUnitRunner])
+class TimerSpec extends TypedSpec("""
+  akka.loglevel = DEBUG
+  """) {
+
+  sealed trait Command
+  case class Tick(n: Int) extends Command
+  case object Bump extends Command
+  case object End extends Command
+  case class Throw(e: Throwable) extends Command
+  case object Cancel extends Command
+
+  sealed trait Event
+  case class Tock(n: Int) extends Event
+
+  class Exc extends RuntimeException("simulated exc") with NoStackTrace
+
+  trait RealTests extends StartSupport {
+    implicit def system: ActorSystem[TypedSpec.Command]
+    implicit val testSettings = TestKitSettings(system)
+
+    val interval = 1.second
+    val dilatedInterval = interval.dilated
+
+    def target(monitor: ActorRef[Event], timer: TimerScheduler[Command], bumpCount: Int): Behavior[Command] =
+      Immutable { (ctx, cmd) ⇒
+        cmd match {
+          case Tick(n) ⇒
+            monitor ! Tock(n)
+            Same
+          case Bump ⇒
+            val nextCount = bumpCount + 1
+            timer.startTimer("T", Tick(nextCount), interval, repeat = true)
+            target(monitor, timer, nextCount)
+          case End ⇒
+            Stopped
+          case Cancel ⇒
+            timer.cancel("T")
+            Same
+          case Throw(e) ⇒
+            throw e
+        }
+      }
+
+    def `must schedule repeated ticks`(): Unit = {
+      val probe = TestProbe[Event]("evt")
+      val behv = timerScheduler[Command] { timer ⇒
```

Comment:
The other behaviours is something the actor is `deferred`, `immutable`, `mutable` while this is something it gets injected. Should it perhaps be `timed` instead, or `withTimer`?